src/pytorch_lightning/loggers/tensorboard.py

```diff
@@ -114,10 +113,6 @@ def __init__(
         sub_dir: Optional[_PATH] = None,
         **kwargs: Any,
     ):
-        if not _TENSORBOARD_AVAILABLE and not _TENSORBOARDX_AVAILABLE:
```

Comment:
This is here to provide a good error message for anybody using this class who's tb/tbX dependency is not working. Otherwise they will only find out when the code actually runs and the local imports are run. We want to fail with a good message as soon as possible

Reply:
can you pls give me a simple example of how this could happen, tbX dependency is not working?
Also why we do not have the same for other mandatory packages like tqdm, fsspec, torchmetrics?

Reply:
> Also why we do not have the same for other mandatory packages like tqdm, fsspec, torchmetrics?

We want to do the same eventually. It's just work in progress tracked in #12786

> how this could happen, tbX dependency is not working?

tbX could not be available for other reasons other than that the dependency is missing. For example, it would show as unavailable if it requires a different dependency that wasn't properly installed. We had an instance of this happen to the NeptuneLogger previously: https://github.com/Lightning-AI/lightning/issues/14695#issuecomment-1247899659

Reply:
It could because a package can appear as installed but still fail the `pkg_resources.require("the_requirement")` test. One example is in the error reported in #14695.

The intention here is to be extra safe and provide a good error message for the user if this happens.

Reply:
but you are referring to extra not mandatory dependency, in such case you shall path torch :otter:

Reply:
tb is functionally an extra dependency, one can use PL without tensorboard installed. Given the problematic nature of it's installation, I can see users choosing to drop it. Either after a pip installation or by modifying the local requirements in a git clone. In that case, this error helps

Reply:
Personally I see this as a temporary solution. TBX still has dependencies we need to get rid of so it has to go eventually.